src/DataSet/ObjectDataSet.php

```diff
@@ -137,32 +79,65 @@ public function getData(): array
             return $this->object->getData();
         }
 
-        if ($this->data !== null) {
-            return $this->data;
+        if ($this->cachedData !== null) {
+            return $this->cachedData;
         }
 
-        $data = [];
-        foreach ($this->getReflectionProperties() as $name => $property) {
-            $data[$name] = $property->getValue($this->object);
+        $this->cachedData = [];
+        foreach ($this->getCachedReflectionProperties() as $name => $property) {
+            $this->cachedData[$name] = $property->getValue($this->object);
         }
 
-        $this->data = $data;
-
-        return $data;
+        return $this->cachedData;
     }
 
-    private function hasCachedRules(): bool
+    private function getCachedRules(): array
     {
-        return array_key_exists($this->cacheKey, self::$cachedRules);
+        $this->assertRulesAndReflectionPropertiesInCache();
+        return self::$cache[$this->cacheKey]['rules'];
     }
 
-    private function getCachedRules(): array
+    private function getCachedReflectionProperties(): array
     {
-        return self::$cachedRules[$this->cacheKey];
+        $this->assertRulesAndReflectionPropertiesInCache();
+        return self::$cache[$this->cacheKey]['reflectionProperties'];
     }
 
-    private function setCachedRules(array $rules): void
+    // TODO: use Generator to collect attributes
+    private function assertRulesAndReflectionPropertiesInCache(): void
```

Comment:
Too much is going on here. And technically it's not assert. The purpose of assert is just verify the presence of data in cache in this case.

Reply:
Goal - check cache and if empty, then get and cache data. Agree that need to rename method. Can you suggest name?

Reply:
I still think that renaming method is not enough in this case.